### DISCUSSION TITLE
Thanks for contributing to Machine Learning and AI with this repository!

### DISCUSSION THREAD

**Discussion by Darlanio** [ggml-org/llama.cpp#236] [General]
I just want to say Thank You!

I got your solution working and I am happy to confirm that it is working as intended. The quality of text produced is probably a bit lower compared to text produced with the full weights, but the quantized weights saves a lot of space and maybe some processing time.

I would like to see the solution rewritten as C# to better understand it, since I am using C# at work but do not use C++ och C since I was studying programming in the 90s. I do understand this is not the main goal of this project, but in case someone else wants to do this or have already done this - please leave me a note.

Thanks again for this lovely work!

**Comment by bernatvadell** [comment]
Hi @Darlanio 

Maybe I can follow up on PR #77. In this PR we are working to export the functionalities in a library (llama.cpp) that in principle could be imported without much effort in dotnet (C#) with DLLImport.

A hug!

**Reply by Darlanio** [reply]
Yes, that would be very nice!
I will keep an eye on https://github.com/ggerganov/llama.cpp/pull/77 
If I am in the future able to include this in some sort of simple C# program, I will link back to this repository (of course) and let you know where to find that repository (some sort of tutorial on how to use this from C# possibly). Maybe I could even post that repo here instead?
